### DISCUSSION TITLE
Feature request/Idea (litedown authoring in obsidian)

### DISCUSSION THREAD

**Discussion by MichelNivard** [yihui/litedown#46] [Ideas]
Hi @yihui your commented on a github discussion I had with the developer of the obsidian code execution plugin[twibiral/obsidian-execute-code/issues/325]  made me think.  Isn't Obsidian a GREAT fit for litedown authoring?Obsidian is 100% markdown focused, and mostly common mark compatible I think. If you'd tweak the existing code execution plugin to generate slightly more pleasing output in visual mode, and you  make custom buttons to render any page to pdf/html (I made those work for Quarto, which was a hassle because quarto doesn't like .md, litedown would have a problem with .md to begin with), yo;'d have a fully featured "literate programming" authoring experience inside an amazing note taking app. This really almost instantly becomes the ultimate data sketchpad so to speak.

With positron looking more like an IDE and far less WYSIWYG "literate programming" than RStudio (visually..) there is space for a tool that abstracts away loads of the developer bells and niceties to fully centrist the "literate programming" note/document.

**Comment by jangorecki** [comment]
I am not against, but worth to note that obsidian is not open source.

**Comment by MichelNivard** [comment]
That does bug me as well, their philosophy of notes kept in .md, making the content independent of the software is great, the closed source nature obviously is not. There are many alternatives some fully closed clearly more restrictive in terms of whether you can walk away with your notes, some entirely open source but with far smaller user bases. Litedown would only fit tools that use .md authoring and support external plugins that could interface with the Users R install. 

 obviously hugely unconvincing to build multiple authoring extensions, for Obsidian it’s at least proven interactive code chunks are feasible, just needs some love/attention to make it really shine. 
 
Given litedown is open source it’s something you could leave to the community (I lack the typescript skill). though I could  see how diversifying the set of WYSIWYG litedown authoring options beyond RStudio (which might eventually be abandoned) and into authoring within .md based notekeeping tools (natural fit I feel) could be an amazing use of litedown.

(FYI I am in no way affiliated with Obsidian, just a user who feels it integrated with litedown it would be the ultimate (data) science sketch pad…

**Comment by yihui** [comment]
I have never used Obsidian, but earlier this year someone mentioned to me that they and I perhaps have one thing in common: minimalism. So I'm definitely curious. I'll try to learn more about it when I have time. Thanks for the suggestion!

**Comment by MichelNivard** [comment]
So just wanted to circle back, I am currently working on an R integration for Obsidian. I feel it's a great environment for pure literate programming. Currently it uses Quarto for rendering out pdf/html. Given I am very ambivalent about python/Julia support (just ambivalent about being able to pull it off I rarely work in python/Julia hard for me to support) I might move to litedown, because it wouldn't raise any expectations wrt other languages I think. Litedown is R only right? Is Litedown currently table and is the syntax expected to remain as is across versions? 

A 30 sec video of the current state, given your preference for minimalism the flashy button ribbon might be a bit too much but its entirely optional!

https://github.com/user-attachments/assets/4f680ac1-6990-41f1-9e27-3a71ff8cbb86
